pkg/gitproviders/provider_org.go

```diff
@@ -0,0 +1,142 @@
+package gitproviders
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/fluxcd/go-git-providers/gitprovider"
+)
+
+type orgGitProvider struct {
+	domain   string
+	provider gitprovider.Client
+}
+
+var _ GitProvider = orgGitProvider{}
+
+func (p orgGitProvider) RepositoryExists(name string, owner string) (bool, error) {
+	orgRef := gitprovider.OrgRepositoryRef{
+		OrganizationRef: gitprovider.OrganizationRef{Domain: p.domain, Organization: owner},
+		RepositoryName:  name,
+	}
+	if _, err := p.provider.OrgRepositories().Get(context.Background(), orgRef); err != nil {
+		if errors.Is(err, gitprovider.ErrNotFound) {
+			return false, nil
+		}
+
+		return false, fmt.Errorf("could not get verify repository exists  %w", err)
+	}
+
+	return true, nil
+}
+
+func (p orgGitProvider) DeployKeyExists(owner, repoName string) (bool, error) {
+	ctx := context.Background()
+
+	orgRepo, err := p.getOrgRepo(owner, repoName)
+	if err != nil {
+		return false, fmt.Errorf("error getting org repo reference for owner %s, repo %s, %s", owner, repoName, err)
+	}
+
+	return deployKeyExists(ctx, orgRepo)
```

Comment:
:thinking: It's weird to me that this works. An `orgRepo` satisfies a `gitprovider.UserRepository` interface? I guess this is down to the way `go-git-providers` works, but still confused as to why they are separate types.